CONTRIBUTING.md

```diff
@@ -354,6 +354,33 @@ static_assert(std::is_same(A*, decltype(A::singelton()))::value, "hmm");
   are too large.  Splitting such files into separate files helps.
   (Example: `THTensorMath`, `THTensorMoreMath`, `THTensorEvenMoreMath`.)
 
+### Running Clang-Tidy
+
+[Clang-Tidy](https://clang.llvm.org/extra/clang-tidy/index.html) is a C++
+linter and static analysis tool based on the clang compiler. We run clang-tidy
+in our CI to make sure that new C++ code is safe, sane and efficient. See our
+[.travis.yml](https://github.com/pytorch/pytorch/blob/master/.travis.yml) file
+for the simple commands we use for this.
+
+To run clang-tidy locally, follow these steps:
+
+1. Install clang-tidy. First, check if you already have clang-tidy by simply
+writing `clang-tidy` in your terminal. If you don't yet have clang-tidy, you
+should be able to install it easily with your package manager, e.g. by writing
+`apt-get install clang-tidy` on Ubuntu. See https://apt.llvm.org for details on
+how to install the latest version. Note that newer versions of clang-tidy will
+have more checks than older versions. In our CI, we run clang-tidy-6.0.
+
+2. Use our driver script to run clang-tidy over your git revision (you may want
+   to replace `HEAD~1` with `HEAD` to pick up uncommitted changes):
+  ```sh
+  $ python tools/clang_tidy.py -d /path/to/build/ -p torch/csrc -r HEAD~1
+  ```
+
+Above, it is assumed you are in the PyTorch root folder. `path/to/build` should
```

Comment:
Most of us are building with `setup.py`, right? Give the instructions for that case, if they're the most likely.

Reply:
You still get a `/path/to/build` right?

Reply:
Err, isn't the default build directory `build`?

Reply:
As in, why do you need to give an absolute path.